src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/DB2Engine.java

```diff
@@ -425,17 +426,92 @@ protected MappedEntity createPreparedStatementForInserts(final DbEntity entity)
         logger.trace(insertStatement);
         logger.trace(insertReturnStatement);
 
-        PreparedStatement ps, psReturn, psWithAutoInc;
+        PreparedStatement ps = null, psReturn = null, psWithAutoInc = null, psUpsert;
         try {
 
             ps = conn.prepareStatement(insertStatement);
             psReturn = conn.prepareStatement(insertReturnStatement);
             psWithAutoInc = conn.prepareStatement(insertWithAutoInc);
 
-            return new MappedEntity().setInsert(ps).setInsertReturning(psReturn).setInsertWithAutoInc(psWithAutoInc).setAutoIncColumn(returning);
+            final String upsert = buildUpsertStatement(entity, columns, values);
+            psUpsert = conn.prepareStatement(upsert);
+
+            return new MappedEntity()
+                        .setInsert(ps)
+                        .setInsertReturning(psReturn)
+                        .setInsertWithAutoInc(psWithAutoInc)
+                        .setAutoIncColumn(returning)
+                        .setUpsert(psUpsert);
+
+        } catch (final IllegalArgumentException e) {
+            logger.trace("Returning entity without an UPSERT/MERGE prepared statement.");
```

Comment:
This logger should be an Error, or at lest a Warning, right?
and the `e` exception should be logged as well.